### PR TITLE
Regex lookaround information updated

### LINKED ISSUE #22282: Description of Zero-width lookbehind/lookahead assertion is missing/wrong
In the documentation, the current example provided for the Zero-width negative *lookahead* assertion gives the results of a Zero-width negative *lookbehind* assertion, which is wrong.

Moreover, there is no description on these within the Grouping Constructs section of the documentation, and from the examples provided in the documentation it doesn't get clear how _lookaround_ assertions behave in particular.

So, I propose to revise the examples of these four grouping elements by utilizing one uniform regular expression pattern and one uniform probe string for all the four groupings, so it gets clear what their particular features are.

**This is my proposal:**

<table>
  <thead>
    <tr>
      <th>Grouping construct</th>
      <th>Description</th>
      <th>Pattern</th>
      <th>Matches</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>

 `(?=`<br/>_subexpression_<br/>`)`
</td>
      <td>

Zero-width positive lookahead assertion.
</td>
      <td>

`\b\w+\b(?=.+and.+)`<br/>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</td>
      <td>

`"cats"`, `"dogs"`<br/>in<br/>`"cats, dogs and some mice."`
</td>
    </tr>
    <tr>
      <td>

`(?!`<br/>_subexpression_<br/>`)`
</td>
      <td>

Zero-width negative lookahead assertion.
</td>
      <td>

`\b\w+\b(?!.+and.+)`<br/>&nbsp;
</td>
      <td>

`"and"`, `"some"`, `"mice"`<br/>in<br/>`"cats, dogs and some mice."`
</td>
    </tr>
    <tr>
      <td rowspan="2">

`(?<=`<br/>_subexpression_<br/>`)`
</td>
      <td rowspan="2">

Zero-width positive lookbehind assertion.
</td>
      <td>

`\b\w+\b(?<=.+and.+)`<br/>&nbsp;
</td>
      <td>

`"some"`, `"mice"`<br/>in<br/>`"cats, dogs and some mice."`
</td>
    </tr>
    <tr>
      <td>

`\b\w+\b(?<=.+and.*)`<br/>&nbsp;
</td>
      <td>

`"and", "some"`, `"mice"`<br/>in<br/>`"cats, dogs and some mice."`
</td>
    </tr>
    <tr>
      <td rowspan="2">

`(?<!`<br/>_subexpression_<br/>`)`
</td>
      <td rowspan="2">

Zero-width negative lookbehind assertion.
</td>
      <td>

`\b\w+\b(?<!.+and.+)`<br/>&nbsp;
</td>
      <td>

`"cats"`, `"dogs"`, `"and"`<br/>in<br/>`"cats, dogs and some mice."`
</td>
    </tr>
    <tr>
      <td>

`\b\w+\b(?<!.+and.*)`<br/>&nbsp;
</td>
      <td>

`"cats"`, `"dogs"`<br/>in<br/>`"cats, dogs and some mice."`
</td>
    </tr>
  </tbody>
</table>

**Please note:**

- To my proposal I only added the four rows in need of improvement. You need to add the other rows from the original table, too, if you'd decide to revise your table.

   If you want me to provide a full Grouping Constructs table, please drop me a note.

- GitHub seems to wrap table columns, even when they contain pre-formatted text.

   So, in order to keep the regular expressions in my proposed content from being accidentally wrapped, I added a number of non-breaking spaces below the first regular expression. So, the regular expressions column won't be wrapped now. To the other table rows I then added a line-break alone, so the vertical spacing appears the same across all table rows.

- In order to provide two row-spanning columns I had to use the HTML table format in favour of the Markdown format because Markdown doesn't support `rowspan`.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 16d4698e-4220-c422-a554-8a481803d240
* Version Independent ID: f0e3b82a-8d81-ad9e-97fc-efb92f7d0d38
* Content: [Regular Expression Language - Quick Reference](https://docs.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-language-quick-reference#grouping-constructs)
* Content Source: [docs/standard/base-types/regular-expression-language-quick-reference.md](https://github.com/dotnet/docs/blob/master/docs/standard/base-types/regular-expression-language-quick-reference.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @adegeo
* Microsoft Alias: **adegeo**

### PR DESCRIPTION
**Pull request by SetTrend**
Information in the regular expression grouping table added as discussed in https://github.com/dotnet/docs/issues/22282#issuecomment-764762593

## Summary

The original list of regular expression grouping constructs was misleading. The samples didn't convey the peculiarities of each grouping construct. Without further reference information the information in the table wasn't helpful.

In the new version the table was updated as described in issue #22282. Significant samples have been provided and brief additional information regarding _lookaround_ and _atomic_ grouping constructs has been added.

I'm not sure if I met all the coding style guides required for the docs. Please advise if something needs improvement.

Fixes #22282

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=22520) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=22520) before we can accept your contribution.<br/><br/>:x: SetTrend [sign now](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=22520)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/docs?pullRequest=22520) it.</sub>

**Comment by SetTrend** [comment]
@adegeo:

I have updated the regular expression grouping construct table as discussed. Please advise if things need to be amended.

**Comment by adegeo** [comment]
Thank you @SetTrend for the effort here! The table however seems to have been changed to an HTML table which makes it harder to maintain/review. Can you swap style back?

**Review by adegeo — CHANGES_REQUESTED** [review]
Thank you @SetTrend for the effort here! The table however seems to have been changed to an HTML table which makes it harder to maintain/review. Can you swap style back?

**Comment by SetTrend** [comment]
Hi @adegeo:

I had to swap to HTML table because Markdown doesn't allow row spans. See my first comment in the original GitHub issue:

> In order to provide two row-spanning columns I had to use the HTML table format in favour of the Markdown format because Markdown doesn't support `rowspan`.

How do you want me to proceed?

**Comment by SetTrend** [comment]
What I did to check for correctness was to paste the page source into a GitHub comment text editor windows (like this one). When clicking on `Preview`, you can perfectly see the result (except for API references, of course).

**Comment by adegeo** [comment]
Ahh I see. I didn't notice it. In general we don't do row spanning just because our markdown syntax doesn't (currently) support this. We've moved on to using tables sparingly and only when we need to. For maintainability, we should keep it as markdown (As ugly as that too can be...)

There really isn't a great solution... What about having a full row and repeating the two columns?

**Comment by SetTrend** [comment]
Sure. Let me update the table. I'll reach out here, say, tomorrow, with an update.

**Comment by adegeo** [comment]
No worries, thank you for your contributions!!

**Comment by SetTrend** [comment]
Hello, @adegeo,

I now updated the table so it will have the Markdown format.

I added two rows of `&mdash;` lines to separate the two similar lookaround variations:

table
<br/>

Would you feel comfortable with this version? Or what would you like me to amend?

Cheers,
Axel

**Comment by SetTrend** [comment]
Hello @adegeo, would you mind kindly approving my pull request?

**Comment by adegeo** [comment]
I'll get on this sorry for the delay. We've been having build issues.

File: docs/standard/base-types/regular-expression-language-quick-reference.md [hunk 1]
```diff
@@ -92,11 +92,25 @@ Grouping constructs delineate subexpressions of a regular expression and typical
...
+|`(?!` *subexpression* `)`|Zero-width negative lookahead assertion.|`\b\w+\b(?!.+and.+)`|`"and"`, `"some"`, `"mice"`<br/>in<br/>`"cats, dogs and some mice."`|
+|`(?<=` *subexpression* `)`|Zero-width positive lookbehind assertion.|`\b\w+\b(?<=.+and.+)`<br/><br/>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br/><br/>`\b\w+\b(?<=.+and.*)`|`"some"`, `"mice"`<br/>in<br/>`"cats, dogs and some mice."`<br/>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br/>`"and"`, `"some"`, `"mice"`<br/>in<br/>`"cats, dogs and some mice."`|
+|`(?<!` *subexpression* `)`|Zero-width negative lookbehind assertion.|`\b\w+\b(?<!.+and.+)`<br/><br/>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br/><br/>`\b\w+\b(?<!.+and.*)`|`"cats"`, `"dogs"`, `"and"`<br/>in<br/>`"cats, dogs and some mice."`<br/>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br/>`"cats"`, `"dogs"`<br/>in<br/>`"cats, dogs and some mice."`|
+|`(?>` *subexpression* `)`|Atomic group.|`(?>a\|ab)c`|`"ac"` in`"ac"`<br/><br/>_nothing_ in`"abc"`|
+
+**Lookarounds At A Glance**
```
Comment (adegeo) [inline docs/standard/base-types/regular-expression-language-quick-reference.md:101]:
```suggestion
### Lookarounds at a glance
```

**Comment by SetTrend** [comment]
Hello @adegeo, thank you for taking the time.

I'm not quite sure about the current status of this PR. You pushed and resolved your requested change yourself (excellent!). However, the PR's status still reads **"Changes requested"**. It looks like the current status refers to this comment:

/dotnet/docs/pull/22520#pullrequestreview-577820505

**Comment by adegeo** [comment]
Still working on it 😄  The builds weren't showing up and after about an hour of waiting I had moved on to something else. I'm still here though 😄

**Review by adegeo — APPROVED** [review]
Looks good and gets the job done. Sorry for the long turnaround on this. We really appreciate you being available when we're not sometimes! The odd thing that happened for almost 2 weeks was the build system died and we couldn't get anything through. That came online yesterday (or the day before?) but went back offline for a bit. Seems it's stable now 🎉

**Comment by SetTrend** [comment]
Thank you so much, @adegeo, for all the great efforts you're undertaking and particularly the efforts you've put into this PR 👍👍👍

I hope the build system will get rid of its bad cough soon 😉

**Pull request merged**